core/drivers/atmel_wdt.c

```diff
@@ -244,29 +246,46 @@ static TEE_Result wdt_node_probe(const void *fdt, int node,
 
 	it = dt_get_irq_type_prio(fdt, node, &irq_type, &irq_prio);
 	if (it == DT_INFO_INVALID_INTERRUPT)
-		goto err_free_wdt;
+		goto err_free;
 
-	it_hdlr = itr_alloc_add_type_prio(it, &atmel_wdt_itr_cb, 0, wdt,
-					  irq_type, irq_prio);
-	if (!it_hdlr)
-		goto err_free_wdt;
+	it_hdlr = calloc(1, sizeof(*it_hdlr));
```

Comment:
It would be nice with an alternative to `interrupt_alloc_add_handler()` which also takes `irq_type` and `irq_prio` into account. I wouldn't mind an `interrupt_alloc_add_handler2()` to take care of that, or `interrupt_alloc_add_conf_handler()` if you don't mind even longer function names.